### PR TITLE
pidgin: 2.13.0 -> 2.14.1

### LINKED ISSUE #95296: Update pidgin to 2.14
Looks like there has been a new release this year:

* https://pidgin.im/posts/2020-06-2.14.0-released/
* https://pidgin.im/posts/2020-06-2.14.1-released/

maintainer: @vcunat 

cc @masipcat it adds XDG portal support, which might be necessary for screen sharing on phosh and other Wayland clients.

### PR DESCRIPTION
**Pull request by magnetophon**
###### Motivation for this change

Closes #95296

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
```diff
-Build console UI.............. : yes
+Build console UI.............. : no
```
I'm not sure if people used it.  I didn't even know about it :-)
I found out in [the NEWS](https://pidgin.im/posts/2020-06-2.14.0-released/); most likely it's just about packaging the `gnt` library that's been separated since 2.14.  It will surely be tiny, so I suppose it will be better to keep the console UI.

**Review by vcunat — APPROVED** [review]
Seems to work OK.  (I use just simple stuff though)

The console UI starts up again and appears to work as well.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102318` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>pidgin-latex</li>
    <li>pidgin-mra</li>
    <li>pidgin-msn-pecan</li>
    <li>pidgin-osd</li>
    <li>pidgin-otr</li>
    <li>pidgin-sipe</li>
    <li>purple-facebook</li>
    <li>purple-lurch</li>
    <li>purple-matrix</li>
    <li>skype4pidgin</li>
    <li>telegram-purple</li>
    <li>toxprpl</li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>libgnt</li>
    <li>pidgin</li>
    <li>pidgin-carbons</li>
    <li>pidgin-opensteamworks</li>
    <li>pidgin-skypeweb</li>
    <li>pidgin-window-merge</li>
    <li>pidgin-xmpp-receipts</li>
    <li>purple-discord</li>
    <li>purple-hangouts</li>
    <li>purple-plugin-pack</li>
    <li>purple-slack</li>
    <li>purple-vk-plugin</li>
    <li>purple-xmpp-http-upload</li>
    <li>telepathy-haze</li>
  </ul>
</details>

**Comment by vcunat** [comment]
I got no failures 🤷🏽 
```
[29 built, 122 copied (374.5 MiB), 93.9 MiB DL]
https://github.com/NixOS/nixpkgs/pull/102318
26 packages built:
libgnt pidgin pidgin-carbons pidgin-latex pidgin-mra pidgin-msn-pecan pidgin-opensteamworks pidgin-osd pidgin-otr pidgin-sipe pidgin-skypeweb pidgin-window-merge pidgin-xmpp-receipts purple-discord purple-facebook purple-hangouts purple-lurch purple-matrix purple-plugin-pack purple-slack purple-vk-plugin purple-xmpp-http-upload skype4pidgin telegram-purple telepathy-haze toxprpl
```

**Comment by vcunat** [comment]
... so perhaps you could find logs of those failures?

**Comment by SuperSandro2000** [comment]
```
building
gcc -I/usr/include/libpurple -DPURPLE_PLUGINS -DENABLE_NLS -Wall -pthread `pkg-config --cflags glib-2.0 gdk-pixbuf-2.0 libnotify purple dbus-glib-1` -I. -g  -pipe libskype.c -o libskype.so -shared -fPIC -DPIC
make: /bin/sh: No such file or directory
make: *** [Makefile:67: libskype.so] Error 127
```

```
configuring
configure flags: --prefix=/nix/store/b7avxfy9pk8gykm1m2gahadd4ric8y90-telegram-purple-1.3.1
/nix/store/q1zjp9grl4w92qalkdqjs2bj5d0pf8ih-stdenv-linux/setup: ./configure: /bin/sh: bad interpreter: No such file or directory
```

```
:$n' --trace='_AM_DEPENDENCIES:$f::$n' --trace='AM_ENABLE_SHARED:$f::$n' --trace='_LT_AC_PROG_CXXCPP:$f::$n' --trace='AC_LIBTOOL_SYS_HARD_LINK_LOCKS:$f::$n' --trace='LT_SYS_MODULE_PATH:$f::$n' --trace='LT_FUNC_ARGZ:$f::$n' --trace='_LT_AC_PROG_ECHO_BACKSLASH:$f::$n' --trace='AC_LIBTOOL_SYS_OLD_ARCHIVE:$f::$n' --trace='AC_LIBLTDL_CONVENIENCE:$f::$n' --trace='AC_LTDL_DLLIB:$f::$n' --trace='AC_LTDL_SYS_DLOPEN_DEPLIBS:$f::$n' --trace='AM_CONDITIONAL:$f::$n' --trace='LTDL_INIT:$f::$n' --trace='AC_LIBTOOL_OBJDIR:$f::$n' --trace='PKG_CHECK_MODULES:$f::$n' --trace='LT_OUTPUT:$f::$n' --trace='LT_FUNC_DLSYM_USCORE:$f::$n' --trace='AC_ENABLE_FAST_INSTALL:$f::$n' configure.ac |: No such file or directory
autoreconf: aclocal failed with exit status: 1
```

but I am rerunning because this looks like some unrelated failure.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102318` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li>pidgin-msn-pecan</li>
    <li>pidgin-osd</li>
    <li>pidgin-sipe</li>
    <li>purple-facebook</li>
    <li>purple-lurch</li>
    <li>purple-matrix</li>
    <li>telegram-purple</li>
  </ul>
</details>
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>libgnt</li>
    <li>pidgin</li>
    <li>pidgin-carbons</li>
    <li>pidgin-latex</li>
    <li>pidgin-mra</li>
    <li>pidgin-opensteamworks</li>
    <li>pidgin-otr</li>
    <li>pidgin-skypeweb</li>
    <li>pidgin-window-merge</li>
    <li>pidgin-xmpp-receipts</li>
    <li>purple-discord</li>
    <li>purple-hangouts</li>
    <li>purple-plugin-pack</li>
    <li>purple-slack</li>
    <li>purple-vk-plugin</li>
    <li>purple-xmpp-http-upload</li>
    <li>skype4pidgin</li>
    <li>telepathy-haze</li>
    <li>toxprpl</li>
  </ul>
</details>

**Comment by vcunat** [comment]
That's really weird.  I believe `/bin/sh` should be there, both in normal NixOS and in the build sandbox (on any system).

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102318` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>pidgin</li>
    <li>purple-matrix</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libgnt</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102318` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>libgnt</li>
    <li>pidgin</li>
    <li>pidgin-carbons</li>
    <li>pidgin-latex</li>
    <li>pidgin-mra</li>
    <li>pidgin-msn-pecan</li>
    <li>pidgin-opensteamworks</li>
    <li>pidgin-osd</li>
    <li>pidgin-otr</li>
    <li>pidgin-sipe</li>
    <li>pidgin-skypeweb</li>
    <li>pidgin-window-merge</li>
    <li>pidgin-xmpp-receipts</li>
    <li>purple-discord</li>
    <li>purple-facebook</li>
    <li>purple-hangouts</li>
    <li>purple-lurch</li>
    <li>purple-matrix</li>
    <li>purple-plugin-pack</li>
    <li>purple-slack</li>
    <li>purple-vk-plugin</li>
    <li>purple-xmpp-http-upload</li>
    <li>skype4pidgin</li>
    <li>telegram-purple</li>
    <li>telepathy-haze</li>
    <li>toxprpl</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102318` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>pidgin</li>
    <li>purple-matrix</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libgnt</li>
  </ul>
</details>

File: pkgs/development/libraries/libgnt/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  outputs = [ "out" "dev" ] ++ lib.optional docs "devdoc";
+
+  nativeBuildInputs = [ meson ninja pkgconfig ]
+    ++ lib.optionals docs [ gtk-doc docbook-xsl-nons ];
+
+  # python2 is optional but Pidgin does depend on it (and python3 won't be accepted in 2.x)
+  buildInputs = [ glib ncurses libxml2 python2 ];
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgnt/default.nix:35]:
```suggestion
  meta = with lib; {
```

stdenv.lib got recently deprecated.

File: pkgs/development/libraries/libgnt/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      "ncurses_sys_prefix = '/usr'" \
+      "ncurses_sys_prefix = '${lib.getDev ncurses}'"
+  ''
+  + lib.optionalString (!docs) ''
+    sed "/^subdir('doc')$/d" -i meson.build
+  '';
+
+  outputs = [ "out" "dev" ] ++ lib.optional docs "devdoc";
+
+  nativeBuildInputs = [ meson ninja pkgconfig ]
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgnt/default.nix:29]:
```suggestion
  nativeBuildInputs = [ meson ninja pkg-config ]
```

File: pkgs/development/libraries/libgnt/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{ stdenv, lib, fetchurl, meson, ninja, pkgconfig, python2
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgnt/default.nix:1]:
```suggestion
{ stdenv, lib, fetchurl, meson, ninja, pkg-config, python2
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by Luflosi** [comment]
What's the status of this PR?

**Comment by vcunat** [comment]
Stuck.  The update would break darwin build, and I don't have a good means of debugging that.  The only way I could think about was to only update on linux.

**Comment by Luflosi** [comment]
@magnetophon Are you still interested in getting this PR merged? I came up with https://github.com/Luflosi/nixpkgs/commit/6b22ff94336d603f8832a939cd1271895ed7c371, which compiles fine both on macOS and Linux. Feel free to take any of my changes and adapt them for this PR.
I don't actually use pidgin, so I can't really test this well.

**Comment by magnetophon** [comment]
@Luflosi Thanks!  

I don't use pidgin anymore either.
Can someone else do a PR instead?

**Comment by vcunat** [comment]
OK, let me take over in  #121794 (WIP).  I'll cherry-pick your update later, and test it on my x86_64 NixOS.  I'm not really using pidgin too much, but it's running all the time.

**Comment by magnetophon** [comment]
@vcunat Thanks!

**Comment by ony** [comment]
It looks like #121794 didn't included update before merge :cry: and we still have only 2.13.0

**Pull request closed without merging**